base/applications/mspaint/mouse.cpp

```diff
@@ -50,6 +51,13 @@ roundTo8Directions(LONG x0, LONG y0, LONG& x1, LONG& y1)
     }
 }
 
+BOOL nearlyEqualPoints(INT x0, INT y0, INT x1, INT y1)
+{
+    INT cxThreshold = toolsModel.GetLineWidth() + GetSystemMetrics(SM_CXDRAG);
```

Comment:
Wouldn't the GetSystemMetrics result have to be scaled to the current zoom level?

Reply:
The line width can be zoomed. The drag distance can't be.

Reply:
OK. Fixed.

Reply:
Wait. Don't you call nearlyEqualPoints with transformed (zoomed) coordinates? Wouldn't it have to be the other way around, then?
I.e. something like UnZoomed(GetSystemMetrics(....))?

Reply:
Umm, Reverted. `x0`, `y0`, `x1`, `y1` are unzoomed values. So, we have to calculate the threshold values in unzoomed values.

Reply:
`GetSystemMetrics(SM_CXDRAG)` is already an unzoomed value.

Reply:
Maybe the term "unzoomed" is a bit to ambiguous. I was under the assumption that nearlyEqualPoints gets called with coordinates in the image's coordinate system, whereas GetSystemMetrics uses the screen's coordinate system, which means that one of them has to be scaled. Is this assumption correct? If both are in the screen's coordinate system, the line width would have to be scaled, though.

Reply:
In the screen coordinate system, the pixel and the coordinate are 1:1. `GetSystemMetrics(SM_CXDRAG)` is also 1:1. This `nearlyEqualPoints` takes the image's coordinates.

Reply:
It cannot get the accurate scaled coordinates from the image's coordinates.

Reply:
Applied `UnZoomed` to `cxDrag` and `cyDrag`.
![image](https://user-images.githubusercontent.com/2107452/147464896-54003c07-d0b8-4b84-8da7-1e601aefa3d2.png)
LGTM.